pkg/api/run_now.go

```diff
@@ -20,59 +25,52 @@ type CheckErrorMessage struct {
 	Error       string `json:"error"`
 }
 
-// RunCanaryResponse represents the response body for a run now request
-type RunCanaryResponse struct {
-	Total   int                 `json:"total"`
-	Failed  int                 `json:"failed"`
-	Success int                 `json:"success"`
-	Errors  []CheckErrorMessage `json:"errors,omitempty"`
-}
-
-func (t *RunCanaryResponse) FromCheckResults(result []*pkg.CheckResult) {
-	t.Total = len(result)
-	for _, r := range result {
-		if r.Pass {
-			t.Success++
-			continue
-		}
-
-		t.Failed++
-		if r.Error != "" {
-			t.Errors = append(t.Errors, CheckErrorMessage{
-				Description: r.GetDescription(),
-				Error:       r.Error,
-			})
-		}
-	}
-}
-
 func RunCanaryHandler(c echo.Context) error {
-	id := c.Param("id")
-
+	checkID := c.Param("id")
 	duty := c.Request().Context().(dutyContext.Context)
 
-	canaryModel, err := db.FindCanaryByID(duty, id)
+	check, err := db.GetCheck(duty, checkID)
 	if err != nil {
 		return errorResponse(c, err, http.StatusInternalServerError)
+	} else if check == nil {
+		return errorResponse(c, fmt.Errorf("check (%s) was not found", checkID), http.StatusNotFound)
 	}
 
-	if canaryModel == nil {
-		return errorResponse(c, fmt.Errorf("canary with id=%s was not found", id), http.StatusNotFound)
+	var canary *v1.Canary
+	if canaryModel, err := db.FindCanaryByID(duty, check.CanaryID.String()); err != nil {
+		return errorResponse(c, err, http.StatusInternalServerError)
+	} else if canaryModel == nil {
+		return errorResponse(c, fmt.Errorf("canary (%s) was not found", check.CanaryID), http.StatusNotFound)
+	} else if canary, err = canaryModel.ToV1(); err != nil {
+		return errorResponse(c, err, http.StatusInternalServerError)
+	}
+
+	canary.Spec = canary.Spec.KeepOnly(check.Name)
+	canary.Status.Checks = map[string]string{
+		check.Name: check.ID.String(),
 	}
 
-	canary, err := canaryModel.ToV1()
 	ctx := context.New(duty, *canary)
+	results, err := checks.RunChecks(ctx)
 	if err != nil {
 		return errorResponse(c, err, http.StatusInternalServerError)
 	}
 
-	result, err := checks.RunChecks(ctx)
-	if err != nil {
+	for _, result := range results {
+		_ = cache.PostgresCache.Add(pkg.FromV1(result.Canary, result.Check), pkg.CheckStatusFromResult(*result))
+		if err := canaryJobs.FormCheckRelationships(ctx.Context, result); err != nil {
+			ctx.Logger.Named(result.Name).Errorf("error forming check relationships: %v", err)
+		}
+	}
+
+	canaryJobs.UpdateCanaryStatusAndEvent(ctx.Context, *canary, results)
+	canaryJobs.CanaryLastRuntimes.Store(canary.GetUID(), time.Now())
+
+	var response models.CheckSummary
+	if err := ctx.DB().Where("id = ?", checkID).First(&response).Error; err != nil {
```

Comment:
I don't think this will be deterministic - another check could have been inserted I think we can just return the results object